app/Filament/Resources/UserResource/Pages/EditUser.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Filament\Resources\UserResource\Pages;
+
+use App\Filament\Resources\UserResource;
+use App\Site\Models\User;
+use Filament\Resources\Pages\EditRecord;
+
+class EditUser extends EditRecord
+{
+    protected static string $resource = UserResource::class;
+
+    public static function getNavigationLabel(): string
+    {
+        return __('Edit');
+    }
+
+    public function getSubheading(): ?string
+    {
+        /** @var User $record */
+        $record = $this->getRecord();
+
+        return '[' . $record->ID . '] ' . $record->User;
+    }
+
+    protected function getHeaderActions(): array
+    {
+        return [
+        ];
+    }
+}
```

Comment:
Any user with the Administrator role can change Legacy permissions to any level (including Moderator). The old logic restricts things such that the new value must be lower than the logged-in user's current value.

Reply:
good point - maybe making it read-only in the admin panels should be enough?
the plan is to migrate away from those very soon.

Reply:
Removed `Permissions` form input and removed it from activity logs again. Just remembered that it wasn't included in the list of logged fields as it won't be around for very long.